service/dap/server_test.go

```diff
@@ -2905,6 +2915,219 @@ func TestUnupportedCommandResponses(t *testing.T) {
+
+					client.VariablesRequest(1000)
+					args := client.ExpectVariablesResponse(t)
+
+					// Test: set string 'baz'
+					expectVarExact(t, args, 0, "baz", "baz", `"bazburzum"`, noChildren)
+					client.SetVariableRequest(1000, "baz", `"BazBurZum"`)
+
+					// After setting a variable, StoppedEvent must be sent
+					// so the client can request new stack trace, and scopes info.
+					client.ExpectStoppedEvent(t)
+
+					if got, want := client.ExpectSetVariableResponse(t), `"BazBurZum"`; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q}", got, want)
+					}
+
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					// Check 'baz' was changed.
+					client.VariablesRequest(1000)
+					args2 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, args2, 0, "baz", "baz", `"BazBurZum"`, noChildren)
+
+					// Test: try to set string type of composite type var 'bar'. That is supported.
+					barRef := expectVarExact(t, args, 1, "bar", "bar", `main.FooBar {Baz: 10, Bur: "lorem"}`, hasChildren)
+					client.SetVariableRequest(barRef, "Bur", `"ipsum"`)
+					client.ExpectStoppedEvent(t)
+					if got, want := client.ExpectSetVariableResponse(t), `"ipsum"`; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q", got, want)
+					}
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					client.VariablesRequest(1000)
+					args3 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, args3, 1, "bar", "bar", `main.FooBar {Baz: 10, Bur: "ipsum"}`, hasChildren)
+
+					// Test: set integer 'a2'
+					client.VariablesRequest(1001)
+					locals := client.ExpectVariablesResponse(t)
+					expectVarExact(t, locals, -1, "a2", "a2", "6", noChildren)
+					client.SetVariableRequest(1001, "a2", "42")
+					client.ExpectStoppedEvent(t)
+
+					if got, want := client.ExpectSetVariableResponse(t), "42"; got.Success != true || got.Body.Value != want {
+						t.Errorf("got %#v, want {Success=true, Body.Value=%q}", got, want)
+					}
+
+					handleStop(t, client, 1, "main.foobar", startLineno)
+
+					client.VariablesRequest(1001)
+					locals2 := client.ExpectVariablesResponse(t)
+					expectVarExact(t, locals2, -1, "a2", "a2", "42", noChildren)
+
+					// Test: set integer 'a2' with an invalid value.
+					client.SetVariableRequest(1001, "a2", "false")
+					if got, want := client.ExpectErrorResponse(t), "can not convert false constant to int"; !strings.Contains(got.Body.Error.Format, want) {
+						t.Errorf("got %#v, want error string containing %q", got, want)
+					}
+
+					// TODO(hyangah): test with an integer field of a global variable (struct type).
+				},
+			}, {
+				// Stop at second breakpoint.
+				execute: func() {
+					handleStop(t, client, 1, "main.barfoo", -1)
+					// Test: set string 'a1' in main.barfoo.
+					// This shouldn't affect 'a1' in main.foobar -
+					// we will check that in the next breakpoint.
```

Comment:
You don't need to go to next breakpoint to check this. You can just load parent scope with a scopes request. That's what happens when users click on it in the UI to view all the vars. Then once you do that, you can try to set it again. That should impact the topmost scope because that's your default. If you switch back to it, you should see it.